gabbi/reporter.py

```diff
@@ -80,7 +80,8 @@ def addUnexpectedSuccess(self, test):
             self.stream.writeln('\t[unexpected success]')
 
     def getDescription(self, test):
-        name = test.test_data['name']
+        # Chop the test method (test_request) off the test.id().
+        name = test.id().rsplit('.', 1)[0]
```

Comment:
This seems like breaking encapsulation, i.e. ideally that functionality would be exposed by `test`'s class (not sure what that'd be, exactly)?